app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -129,6 +133,51 @@ public void onSlide(View bottomSheet, float slideOffset) {
         bottomSheetBehaviorForDetails.setState(BottomSheetBehavior.STATE_HIDDEN);
     }
 
+    private void initPermissionWarnings() {
+        locationPermissionDeniedAlert = new AlertDialog.Builder(this)
```

Comment:
IMO these alerts can be created in location manager or something so that it can be reused by other activities.